server/src/internalClusterTest/java/org/elasticsearch/features/ClusterFeaturesIT.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.features;
+
+import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasItem;
+
+public class ClusterFeaturesIT extends ESIntegTestCase {
+
+    @SuppressForbidden(reason = "Directly checking node features in cluster state")
+    public void testClusterHasFeatures() {
+        internalCluster().startNodes(2);
+        ensureGreen();
+
+        FeatureService service = internalCluster().getCurrentMasterNodeInstance(FeatureService.class);
+
+        assertThat(service.getNodeFeatures(), hasItem(FeatureService.FEATURES_SUPPORTED.id()));
+
+        // check the nodes all have a feature in their cluster state (there should always be features_supported)
+        var response = clusterAdmin().state(new ClusterStateRequest().clear().nodes(true)).actionGet();
+        var features = response.getState().clusterFeatures().nodeFeatures();
+        Set<String> missing = features.entrySet()
+            .stream()
+            .filter(e -> e.getValue().contains(FeatureService.FEATURES_SUPPORTED.id()) == false)
+            .map(Map.Entry::getKey)
+            .collect(Collectors.toSet());
+        assertThat(missing + " out of " + features.keySet() + " does not have the required feature", missing, empty());
```

Comment:
Could we assert that all nodes have the same features? That's going to be a more useful assertion once we drop support for featureless nodes I think.

Reply:
Good idea - added